lib/archivist/index.js

```diff
@@ -694,7 +694,15 @@ Archivist.prototype.scan = function (topic, partialIndex, options, cb) {
 		}
 
 		const queries = indexes.map((index) => ({ topic, index }));
-		this.mget(queries, options, cb);
+		this.mget(queries, options, (error, data) => {
+			// Add index to data
+			// Eg. [ [{ userId: 1 }, { name: "foo" }], [{ userId: 2 }, { name: "bar" }] ]
+			const indexedData = data.map((obj, idx) => {
+				return [queries[idx].index, obj];
```

Comment:
You could map against the indexes array instead, would make the code shorter I think.

Reply:
Right

Reply:
Right